File: codes/cpp/basics/gcd.cpp
#include<iostream>
int gcd(int max, int min){
	int hcf=max%min;
	while(hcf!=0){
		max=min;
		min=hcf;
		hcf=max%min;
	}
	return min;
}
int main(){
	int a,b;
	std::cin>>a>>b;
	std::cout<<gcd((a>b?a:b),(a<b?a:b));
	return 0;
}

File: codes/java/practice/baiscs/gcd.java
import java.util.Scanner;
public class gcd{
	public static void main(String[] agrs){
		Scanner cin=new Scanner(System.in);
		int n=cin.nextInt();
		int m=cin.nextInt();
		System.out.println(GCD(Integer.max(n,m),Integer.min(n,m)));
	}
	public static int GCD(int a,int b){
		while(a%b!=0){
			int rem=a%b;
			a=b;
			b=rem;
		}
		return b;
	}
} 
